referral_customer.go

```diff
@@ -141,12 +140,17 @@ func (c *Client) AddReferralCustomerCreditCard(referralCustomerApiKey string, cr
 func (c *Client) AddReferralCustomerCreditCardWithContext(ctx context.Context, referralCustomerApiKey string, creditCardOptions *CreditCardOptions, priority PaymentMethodPriority) (out *PaymentMethodObject, err error) {
 	stripeApiKeyResponse, err := c.retrieveEasypostStripeApiKey(ctx)
 	if err != nil || stripeApiKeyResponse == nil || stripeApiKeyResponse.PublicKey == "" {
-		return nil, errors.New("could not create Stripe token, please try again later")
+		return nil, &InternalServerError{
+			APIError: APIError{
+				Code:       "Could not create Stripe token, please try again later",
+				StatusCode: 0,
```

Comment:
question: why a 0 status code here? I don't think that's what we want to be returning to users. Do we have custom logic that transforms that to something else? Why not return an appropriate 5xx error?

Reply:
Because it's not really the response triggered by the HTTP failure, we're manufacturing this. The Status Code isn't shown anywhere, so it's mostly a placeholder. But we can make it 500 just to be consistent with what a InternalServerError would be.